src/cpp/include/openvino/genai/image_generation/clip_text_model_with_projection.hpp

```diff
@@ -3,100 +3,14 @@
 
 #pragma once
 
-#include <filesystem>
-#include <string>
-
-#include "openvino/genai/visibility.hpp"
-#include "openvino/genai/tokenizer.hpp"
-#include "openvino/genai/lora_adapter.hpp"
-
-#include "openvino/core/any.hpp"
-#include "openvino/runtime/tensor.hpp"
-#include "openvino/runtime/infer_request.hpp"
-#include "openvino/runtime/properties.hpp"
+#include "openvino/genai/image_generation/clip_text_model.hpp"
 
 namespace ov {
 namespace genai {
 
-class OPENVINO_GENAI_EXPORTS CLIPTextModelWithProjection {
+class CLIPTextModelWithProjection : public CLIPTextModel {
 public:
-    struct OPENVINO_GENAI_EXPORTS Config {
-        size_t max_position_embeddings = 77;
-        size_t num_hidden_layers = 32;
```

Comment:
`num_hidden_layers` is not the same for `CLIPTextModel`: https://github.com/openvinotoolkit/openvino.genai/blob/de1836fbe69d0b4756e9584c752ec9ed6bb963d8/src/cpp/include/openvino/genai/image_generation/clip_text_model.hpp#L25

Reply:
we read this value from config anyway:
https://github.com/openvinotoolkit/openvino.genai/blob/de1836fbe69d0b4756e9584c752ec9ed6bb963d8/src/cpp/src/image_generation/models/clip_text_model.cpp#L34

Reply:
Not sure if we can get into a situation where the json doesn’t contain `num_hidden_layers` value - in that case, `read_json_param` will silently use the default value
Probably we should add a warning when `num_hidden_layers` is missing